### PR TITLE
Makes it possible to speed up the tram

### PR DESCRIPTION
**Pull request by Thunder12345**
## About The Pull Request

Makes it possible to speed up the tram.

## Why It's Good For The Game

Because it's worth $200 to me, plus more reasons upon which I will elaborate if found to work when testmerged

## Changelog

:cl:
expansion: Trams can now move faster
/:cl:

### CODE REVIEW

File: code/__DEFINES/machines.dm [hunk 1]
```diff
@@ -216,3 +216,6 @@
 #define TLV_OUTSIDE_WARNING_LIMIT 1
 ///the gas is outside the hazard limit, the air alarm will go into hazard mode
 #define TLV_OUTSIDE_HAZARD_LIMIT 2
+
+///the default multiplier for the speed of moving trams
+#define TRAM_DEFAULT_SPEED 3
```
Comment (tralezab) [inline code/__DEFINES/machines.dm:221]: is machines really the right place? After all, it is a structure

**Comment by Kylerace** [comment]
this doesnt fix any of the issues that are stopping the tram from being sped up, ive tried this before and it was reverted here https://github.com/tgstation/tgstation/pull/60537 because the tram takes up over a third of the tick at base and cannot yield throughout any of that, to the point that it chokes out other subsystems going after it since they have much less time to execute in the tick.

**Comment by LemonInTheDark** [comment]
Since kyler is working on the same bounty as you and can't reasonably do it, I'm going to block this myself.
This is not an acceptable speedup for the tram. I will now explain why.

The issue with tram code is not that when firing at max speed it doesn't fire often enough.
The issue is during highpop, it consumes WAY too much of its proportion of the tick just moving once, and so gets delayed by the mc.
What you are doing here bypasses the mc in the same way as kyler's old ssticker pr did, by offloading the work to the timer subsystem, which does not have this delaying behavior.
This will just cause massive visual lag at highpop at worst, or simply outcompete all other subsystems for cpu time at best.

As I mentioned in the bounty thread that I think inspired this pr, https://tgstation13.org/phpBB/viewtopic.php?p=633855#p633855, the correct thing to do is to make what the tram actually **does** cheaper, not just make it ignore the mc's control.

I'm going to DNM this pr, will be closing at 24 hours.

**Comment by optimumtact** [comment]
this in no way justifies the 200 dollar bounty and actually is an incredible PR to even propose.

**Pull request closed without merging**